domain/src/DomainListBuilder.php

```diff
@@ -13,9 +13,14 @@
 class DomainListBuilder extends DraggableListBuilder {
 
   /**
+   * Provide class-specific value of $entitiesKey.
+   *
    * {@inheritdoc}
    */
-  protected $entitiesKey = 'domains';
```

Comment:
I think I can show that we do not need to declare the property with the code below:

`<?php

abstract class MyParent {
  protected $maskedData = 'masked';

  protected $parentData = 'parent';

  public function __construct() {
  }

  protected function test() {
    echo 'In ', __CLASS__, ": $this->maskedData\n";
    echo 'In ', __CLASS__, ": $this->parentData\n";
  }

  public function reset() {
    $this->maskedData = $this->parentData = 'reset';
  }
}

class MyChild extends MyParent {
  protected $maskedData = 'changed by child definition';

  public function __construct() {
    parent::__construct();
    $this->parentData = 'changed by constructor';
  }

  public function test() {
    parent::test();
    echo 'In ', __CLASS__, ":  $this->maskedData\n";
    echo 'In ', __CLASS__, ":  $this->parentData\n";
  }
}

$child = new MyChild();
$child->test();
$child->reset();
$child->test();
`

I have one had a co-worker get bit by the bug this sort of thing can cause, so I was hoping to be able to demonstrate that as well. But have so far failed. Conceptually, what we want is for there to be one variable, which is part of the motivation - for clarity of code. On the other hand, there are several places in core that do use the original construction. Given that fact, I spent the last hour trying to decide myself if it actually make sense to make this change. It works, I think it seems safe, but being different than core with a solid reason does not seem like a virtue.

I'm going to look at your other comments and come back to this.